pandas/core/base.py

```diff
@@ -311,13 +312,13 @@ def _aggregate(self, arg, *args, **kwargs):
             if _axis != 0:  # pragma: no cover
                 raise ValueError("Can only pass dict with axis=0")
 
-            obj = self._selected_obj
+            selected_obj = self._selected_obj
 
             # if we have a dict of any non-scalars
             # eg. {'A' : ['mean']}, normalize all to
             # be list-likes
             if any(is_aggregator(x) for x in arg.values()):
-                new_arg = {}
+                new_arg: Dict[Label, Union[AggFuncTypeBase, List[AggFuncTypeBase]]] = {}
```

Comment:
Orthogonal to this PR but seems strange that this is really a subset of `AggFuncType` - any chance we have test coverage for the other parts of that Union coming through this method?

Reply:
I believe you're asking why this can't be typed `Dict[Label, AggFuncType]`. The last part of AggFuncType not included here are where the values are a Dict. This has been removed and will raise on 338 below. That it raises is tested in `frame.apply.test_agg_dict_nested_renaming_depr`.